src/test/scala/org/broadinstitute/dsde/workbench/sam/api/ManagedGroupRoutesSpec.scala

```diff
@@ -0,0 +1,115 @@
+package org.broadinstitute.dsde.workbench.sam.api
+
+import akka.http.scaladsl.model.{StatusCodes, Uri}
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail, WorkbenchUserId}
+import org.broadinstitute.dsde.workbench.sam.{TestSupport, model}
+import org.broadinstitute.dsde.workbench.sam.model._
+import org.broadinstitute.dsde.workbench.sam.service.ManagedGroupService
+import org.scalatest.{BeforeAndAfter, FlatSpec, GivenWhenThen, Matchers}
+
+/**
+  * Created by dvoet on 6/7/17.
+  */
+class ManagedGroupRoutesSpec extends FlatSpec with Matchers with ScalatestRouteTest with TestSupport with BeforeAndAfter {
+
+  private val ownerRoleName = ResourceRoleName("admin")
+  private val ownerPolicyName = AccessPolicyName(ownerRoleName.value)
+  private val memberPolicyName = AccessPolicyName(ManagedGroupService.MemberRoleName.value)
+  private val accessPolicyNames = Set(ownerPolicyName, memberPolicyName)
+  private val policyActions: Set[ResourceAction] = accessPolicyNames.flatMap(policyName => Set(SamResourceActions.sharePolicy(policyName), SamResourceActions.readPolicy(policyName)))
+  private val resourceActions = Set(ResourceAction("delete")) union policyActions
+  private val resourceActionPatterns = resourceActions.map(action => ResourceActionPattern(action.value))
+  private val defaultOwnerRole = ResourceRole(ownerRoleName, resourceActions)
+  private val defaultRoles = Set(defaultOwnerRole, ResourceRole(ManagedGroupService.MemberRoleName, Set.empty))
+  private val managedGroupResourceType = ResourceType(ManagedGroupService.ManagedGroupTypeName, resourceActionPatterns, defaultRoles, ownerRoleName)
+  private val resourceTypes = Map(managedGroupResourceType.name -> managedGroupResourceType)
+  private val groupId = "foo"
```

Comment:
I'm usually in favor of inlining these in the tests. Extracting common values implies that usages are somehow linked, that a change in one place necessitates a change in every other place. Since these are used across (independent) tests, this kind of link/interdependency is the opposite of what we want.

To put it another way, a change to the value here should require review of every test that uses it to make sure that the behavior being tested hasn't changed. While slightly more verbose, repeating the instantiation of the value forces that. (Sure, someone could search/replace but the PR would still explicitly declare that it is being changed for multiple test cases and enumerates exactly which test cases are affected.)

In contrast, `ownerRoleName` here is fine because it is a single unique fact about the component being tested rather than data for a test case. There might be some gray area in the values between `ownerRoleName` and `groupId`... I haven't fully analyzed them all... but we can chat in person if you have questions or counter arguments.

Reply:
@breilly2 I totally understand where you're coming from and I think it's definitely a tradeoff.  You'll see in some of the tests, I am creating some new instances of the objects I need, whereas in most of the tests, the defaults are just fine.  

That said, the spec files I wrote are now oddballs that do not conform to the coding conventions you've been using.  If the team would prefer that I rewrite these tests in that explicit style, I am happy to do so.